pyccel/ast/core.py

```diff
@@ -2570,9 +2493,9 @@ def __init__(
         # name
 
         if isinstance(name, str):
-            name = Symbol(name)
-        elif not isinstance(name, Symbol):
-            raise TypeError('Function name must be Symbol or string')
+            name = PyccelSymbol(name)
+        elif not isinstance(name, PyccelSymbol):
+            raise TypeError('Function name must be PyccelSymbol or string')
```

Comment:
This case is equivalent to an `else`, because `PyccelSymbol` is a subclass of `str` now.

Reply:
8092411